docs/ldml/tr35-keyboards.md

```diff
@@ -1348,206 +1343,503 @@ This example shows some of the mappings for a French keyboard layout:
+
+    - …
+    - character `e`
+    - marker `circ_marker`
+
+5. Now, the input context matches the transform.  The `e` and the marker are replaced with `ê`.
+
+    The input context now has:
+
+    - …
+    - character `ê`
+
+**Effect of markers on final text**
+
+All markers must be removed before text is returned to the application from the input context.
+If the input context changes, such as if the cursor or mouse moves the insertion point somewhere else, all markers in the input context are removed.
+
+**Implementation Notes**
+
+Ideally, markers are implemented entirely out-of-band from the normal text stream. However, implementations _may_ choose to map each marker to a [Unicode private-use character](https://www.unicode.org/glossary/#private_use_character) for use only within the implementation’s processing and temporary storage in the input context.
+
+For example, the first marker encountered could be represented as U+E000, the second by U+E001 and so on.  If a regex processing engine were used, then those PUA characters could be processed through the existing regex processing engine.  `[^\uE000-\uE009]` could be used as an expression to match a character that is not a marker, and `[Ee]\u{E000}` could match `E` or `e` followed by the first marker.
+
+Such implementations must take care to remove all such markers (see prior section) from the resultant text. As well, implementations must take care to avoid conflicts if applications themselves are using PUA characters, such as is often done with not-yet-encoded scripts or characters.
+
+* * *
+
+### <a name="Element_transformGroup" href="#Element_transformGroup">Element: transformGroup</a>
+
+> <small>
+>
+> Parents: [transforms](#Element_transforms)
+>
+> Children: [import](#Element_import), [reorder](#Element_reorder), [_special_](tr35.md#special), [transform](#Element_transform)
+>
+> Occurrence: optional, multiple
+> </small>
+
+A `transformGroup` represents a set of transform elements or reorder elements.
+
+Each `transformGroup` is processed entirely before proceeding to the next one.
+
+**Examples**
+
+Each `transformGroup` element must have **one** of the following three forms. It is not allowed to have an empty `<transformGroup>` element.
+
+#### Form 1: `transformGroup`: import-only
+
+This is a `transformGroup` that consists only of [`import`](#element-import) elements. No other elements are allowed, with the exception of [`special`](tr35.md#special). With this form, only `transform` OR `reorder` elements may be imported by all import statements, but not both.
+
+```xml
+<transformGroup>
+    <import path="..."/>
+    <!-- other <import> elements -->
+</transformGroup>
+```
+
+
+#### Form 2: `transformGroup`: transform
+
+This is a `transformGroup` that consists of one or more [`transform`](#element-transform) elements, prefaced by one or more `import` elements. See the discussion of those elements for details.
```

Comment:
I realize I probably missed the discussion day that led to this decision, but... we have an import-only version of transformGroup and others that also allow imports?

The wording for "import-only" groups says that "no other elements are allowed", but here for the 'transform' variant we have a transformGroup (the example below) that looks identical until a "non-allowed" element is found.  How do we plan to validate this?

What is gained by Form 1 that Form 2 doesn't get?  Otherwise... why would I not just target use of Form 2 and not care to learn about Form 1 whatsoever?  Form 1 appears to be a strict subset of Form 2 to me.

Reply:
I think the only thing we want to disallow is both `transform` AND `reorder`.

Originally, `import` was to be disallowed when used with `transform`, but I think it just makes it too complicated.  

I'll reword this. The DTD already says:

```xml
<!ELEMENT transformGroup ( import*, ( transform* | reorder* ), special* ) >
```

Reply:
See if this looks better now.

Reply:
Better, but it's still unclear.  It sounds like the imports here would probably be affected by the same rule as in Group 1... but Group 2 still accepts multiples, so the whole "strict subset" thing still seems to be in effect.

The only thing clarified is the non-heterogenous import rule - which _is_ much clearer now, to be sure.  But... that's not what the original comment was addressing.